src/librbd/ManagedLock.h

```diff
@@ -29,13 +29,19 @@ class ManagedLock {
 public:
   static const std::string WATCHER_LOCK_TAG;
 
+  enum Mode {
```

Comment:
Nit: can you move this to managed_lock/Types.h? I have assigned myself a tracker ticket to remove all unnecessary dependencies to this templated class in order to clean up the unit tests.